.github/workflows/ci_pipeline.yml

```diff
@@ -0,0 +1,1384 @@
+name: ci_pipeline
+
+# All in one ci pipeline
+#  Note that the jobs system-test_offsite_{abcd} , install-and-use-rpi_m32sd10{abcd}
+#  are there four times and are almost identical. They only differ in a character
+#  a,b,c,d that we use to select a runner.
+#
+# TODO: Reduce complexity by moving functionality to bash scripts
+# TODO: Think about automated generation of this workflow, there is
+#   duplicated code, especially in the later sections
+# TODO: Smoke testing on Azure is disabled, we should check if it works out
+#   with 4 runners in parallel or enable only one
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
+    branches: [main]
```

Comment:
I am wondering if the specifying `main` in `workflow_dispatch` is meaningful. 

P.S. I just realized that `workflow_dispatch` can accept user input and can use them as environmental values. https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch